### DISCUSSION TITLE
load large model

### DISCUSSION THREAD

**Discussion by aiqibaobei** [openscenegraph/OpenSceneGraph#1309] [Q&A] [answered]
Hello, respected author! Thank you for your outstanding contributions. I am currently developing the functionality to load large .ive model files, with file sizes reaching up to approximately 10GB. Despite setting the rendering mode to CullThreadPerCameraDrawThreadPerContext, the browsing experience still feels somewhat laggy. Is there a way to achieve smooth navigation for large model files? Thank you!

**Comment by robertosfield** [comment, marked as answer]
You need to port your application to the [VulkanSceneGraph](https://github.com/vsg-dev/VulkanSceneGraph) if you care about performance, both the scene graph itself is much faster than the OpenSceneGraph and the graphics API Vulkan is much faster than OpenGL.

For extremely large models both scene graphs require you to use database paging to load balance the system, this allows you to handle terabyte sized models easily.  Both scene graphs support PagedLOD/DatabasePager.

However, you'll be held back for getting the most performance out of your hardware is you use the OSG so I recommend rather than consume time trying to eck out a few extra % from the OSG you are better spending your time porting to the VSG and see significant improvement in performance - I often see 3-4 x faster, some models are even 20x faster with VSG/Vulkan.

**Comment by aiqibaobei** [comment]
OK，thank you so much！
